pkg/tessera/tessera.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2025 The Sigstore Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tessera
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	rekor_pb "github.com/sigstore/protobuf-specs/gen/pb-go/rekor/v1"
+	"github.com/sigstore/rekor-tiles/pkg/signer"
+	logformat "github.com/transparency-dev/formats/log"
+	tessera "github.com/transparency-dev/trillian-tessera"
+	"github.com/transparency-dev/trillian-tessera/client"
+	"github.com/transparency-dev/trillian-tessera/storage/gcp"
+)
+
+type Storage struct {
+	origin     string
+	awaiter    *tessera.IntegrationAwaiter
+	addFn      tessera.AddFn
+	readTileFn client.TileFetcherFunc
+}
+
+func NewStorage(ctx context.Context, origin, bucket, spanner, keyPath, keyPass string) (*Storage, error) {
```

Comment:
Does it make sense for this be named according to the backend type? How would we add other backend types?

Could bucket and spanner be abstracted interfaces for leaf storage and metadata storage?

Reply:
I think it still makes sense for this to be `Storage` and `NewStorage` because the object is going to work the same no matter what the backend is. I'll change this to accept a `tessera.Driver` instead of a bucket and spanner string so the caller just has to create and pass in the driver, and it will work the same.